core/lib/types/src/tx/tx_execution_info.rs

```diff
@@ -33,21 +35,35 @@ impl TxExecutionStatus {
 
 #[derive(Debug, Default, Clone, Copy, PartialEq)]
 pub struct DeduplicatedWritesMetrics {
+    // The number of initial storage writes.
     pub initial_storage_writes: usize,
+    // The number of repeated storage writes.
     pub repeated_storage_writes: usize,
+    // This is the total number of bytes used for value updates as part of storage writes.
+    pub total_updated_values_size: usize,
 }
 
 impl DeduplicatedWritesMetrics {
     pub fn from_tx_metrics(tx_metrics: &TransactionExecutionMetrics) -> Self {
         Self {
             initial_storage_writes: tx_metrics.initial_storage_writes,
             repeated_storage_writes: tx_metrics.repeated_storage_writes,
+            total_updated_values_size: tx_metrics.total_updated_values_size,
         }
     }
 
-    pub fn size(&self) -> usize {
-        self.initial_storage_writes * InitialStorageWrite::SERIALIZED_SIZE
-            + self.repeated_storage_writes * RepeatedStorageWrite::SERIALIZED_SIZE
+    pub fn size(&self, protocol_version: ProtocolVersionId) -> usize {
+        match protocol_version {
+            ProtocolVersionId::Version17 => {
+                self.total_updated_values_size
+                    + (BYTES_PER_DERIVED_KEY as usize) * self.initial_storage_writes
+                    + (BYTES_PER_ENUMERATION_INDEX as usize) * self.repeated_storage_writes
+            }
+            _ => {
+                self.initial_storage_writes * InitialStorageWrite::SERIALIZED_SIZE
+                    + self.repeated_storage_writes * RepeatedStorageWrite::SERIALIZED_SIZE
+            }
+        }
```

Comment:
It's better to do `if protocol_version >= 17 then ... else ...`, so for versions > 17 we apply new impl